src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminAuditService.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.dts.fact.entity.Audit;
+import uk.gov.hmcts.dts.fact.repositories.AuditRepository;
+import uk.gov.hmcts.dts.fact.repositories.AuditTypeRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@SuppressWarnings("PMD.UseObjectForClearerAPI")
+public class AdminAuditService {
+
+    private final AuditRepository auditRepository;
+    private final AuditTypeRepository auditTypeRepository;
+
+    @Autowired
+    public AdminAuditService(AuditRepository auditRepository, AuditTypeRepository auditTypeRepository) {
+        this.auditRepository = auditRepository;
+        this.auditTypeRepository = auditTypeRepository;
+    }
+
+    public List<uk.gov.hmcts.dts.fact.model.admin.Audit> getAllAuditData() {
+        return auditRepository
+            .findAll()
+            .stream()
+            .map(audit -> new uk.gov.hmcts.dts.fact.model.admin.Audit(
+                audit.getId(),
+                audit.getUserEmail(),
+                audit.getAuditType(),
+                audit.getActionDataBefore(),
+                audit.getActionDataAfter(),
+                audit.getLocation(),
+                audit.getCreationTime()
+            )).collect(Collectors.toList());
+    }
+
+    public void saveAudit(String auditType, String auditDataBefore, String auditDataAfter) {
+        auditRepository.save(new Audit(
+            SecurityContextHolder.getContext().getAuthentication().getName(),
+            auditTypeRepository.findByName(auditType),
+            auditDataBefore,
+            auditDataAfter,
+            LocalDateTime.now()
```

Comment:
Does it matter that the time recorded in the audit table will be different to the actual time the entity changed (albeit by a small amount)?

Reply:
The time is the creation time :) So it should be fine. I don't think we are updating audits atm, since we prob want to store what happened previously